portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,8 +55,25 @@ public void init() {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
 
-    String json = convertToJson(comments);
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(name, text, timestamp);
+      comments.add(comment);
+    }
+
+    int limit = getLimit(request);
```

Comment:
Two questions:
- It took me a second to understand what the limit was. Could we name this more clearly?
- It looks like we're getting all the comments out of the database, and then dropping anything other than the last `n`. That'll be slow if there's a lot of comments. Instead, could we update the query above based on the limit?

Reply:
I will rename the variable to a more suitable name. And yes I see that that is a more efficient way of doing it. I will try to implement that now.